airflow_prometheus_exporter/prometheus_exporter.py

```diff
@@ -407,22 +440,41 @@ def sla_check(sla_interval, sla_time, max_execution_date, cadence, execution_dat
 def get_sla_miss_dags():
```

Comment:
just an observation - kind of seems like `get_sla_miss_dags` and `get_sla_miss_tasks` are largely pretty similar.
would be nice if you could combine those 2 functions into 1 somehow that way if the logic changes in the future, we wouldnt need to maintain 2 versions of the code

Reply:
Yea, I really want to clean this up.

Reply:
no worries if its too much effort